### ISSUE TITLE
test: Missing argument at index 2

### ISSUE THREAD

**Issue by mchwalisz** [fisherman/autoenvstack#6]
I'm pretty sure it is connected to #3 

When I start terminal in folder without `.env.fish` everything works correctly. If I start in `.env.fish` after #5, the env.fish is sourced but on changed folder I get:

```
test: Missing argument at index 2
```

I assume there is a problem with reverting environment state when `.env.fish` was sourced at start.

**Comment by mchwalisz** [comment]
Small update on the issue. I did some digging and for me it looks like autoenvstack reverts much more variables than it should if it is started at boot. This makes other plugins fail randomly if they where loaded AFTER (I assume) autoenvstack, as all their variables will be cleaned on folder change.

**Comment by hutenosa** [comment]
That's correct -- autoenvstack is quite agressive, when leaving the directory with `.env.fish` it restores the state of the parent directory and removes all variables that are found in its current state. It is made intentionally -- `.env.fish` could be any script, not necessarily the list of key-value pairs.  

This is why if something is set after autoenvstack saved its state, it will be later removed, since there's no way to distinguish the variables set in `.env.fish` from those set manually and by plugins.

There could be many solutions to this issue, but I could not find the cleanest one. Probably the following one is the least annoying:

On leaving a directory with `.env.fish` file, check if a file `.unenv.fish` is present. If yes, execute it, instead of brutally cleaning up the current state. 

The only disadvantage is that would require users to keep in sync both env and unenv files.

Do you agree with this change, @mchwalisz Or you have some other ideas?

**Comment by mchwalisz** [comment]
I see two other sensible solutions (don't know how to implement them yet):

1. Store only changes made by `.env.fish` and restore just those, leaving all other untouched. If I understand you correctly if I declare new variable in a folder with `.env.fish` it will be gone if I leave it. I had the impression that only changes made by `.env.fish` should be reverted. 

2. Do `.env.fish` magic at the end of all loading, so it still could have current behavior. There needs to be a split between initialization and `.env.fish` loading.

IMO the first option is better.

**Comment by ryanartecona** [comment]
I am also affected by this issue. I agree the current behavior as of #5 is a problem.

At the very least, a method of opting out of the automatic `_autoenvstack` at startup could be provided even if it is kept by default. A universal env var could be set to disable it perhaps.

Relatedly, I've been using autoenvstack for several months with a manual `_autoenvstack` at the bottom of my config.fish. I had automatic env stack init before #5 working perfectly. After #5, if I started a shell in a directory with a `.env.fish`, it would work ok until I `cd ..`, at which point it would trash all the env vars I set in my config.fish, rendering my shell barely usable.
